x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/v2/RollupActionSingleNodeTests.java

```diff
@@ -404,7 +405,46 @@ private void assertRollupIndex(RollupActionConfig config, String sourceIndex, St
         SearchResponse resp = client().prepareSearch(rollupIndex).setTrackTotalHits(true).get();
         assertThat(resp.getHits().getTotalHits().value, equalTo(numBuckets));
 
-        // assert that temporary index was removed
+        GetIndexResponse indexSettingsResp = client().admin().indices().prepareGetIndex().addIndices(sourceIndex, rollupIndex).get();
+        // Assert rollup metadata are set in index settings
+        assertEquals(indexSettingsResp.getSetting(sourceIndex, "index.uuid"),
+            indexSettingsResp.getSetting(rollupIndex, "index.rollup.source.uuid"));
+        assertEquals(indexSettingsResp.getSetting(sourceIndex, "index.provided_name"),
+            indexSettingsResp.getSetting(rollupIndex, "index.rollup.source.name"));
+
+        // Assert field mappings
+        Map<String, Map<String,Object>> mappings = (Map<String, Map<String, Object>>) indexSettingsResp.getMappings().get(rollupIndex)
+            .getSourceAsMap().get("properties");
+
+        RollupActionDateHistogramGroupConfig dateHistoConfig = config.getGroupConfig().getDateHistogram();
+        assertEquals(DateFieldMapper.CONTENT_TYPE, mappings.get(dateHistoConfig.getField()).get("type"));
+        Map<String, Object> dateTimeMeta = (Map<String, Object>) mappings.get(dateHistoConfig.getField()).get("meta");
+        assertEquals(dateHistoConfig.getTimeZone(), dateTimeMeta.get("time_zone"));
+        assertEquals(dateHistoConfig.getInterval().toString(), dateTimeMeta.get(dateHistoConfig.getIntervalTypeName()));
+
+        for (MetricConfig metricsConfig : config.getMetricsConfig()) {
+            assertEquals("aggregate_metric_double", mappings.get(metricsConfig.getField()).get("type"));
+            // TODO: Break avg into sum + value_count
```

Comment:
is this TODO for this PR, or a follow-up?

Reply:
Sorry I missed that. Now committed